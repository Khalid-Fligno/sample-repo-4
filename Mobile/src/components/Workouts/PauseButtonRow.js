import React from "react";
import PropTypes from "prop-types";
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Dimensions,
} from "react-native";
import Icon from "../../components/Shared/Icon";
import fonts from "../../styles/fonts";
import colors from "../../styles/colors";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";

const { width } = Dimensions.get("window");
export default class PauseButtonRow extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    const {
      nextExerciseName,
      handlePause,
      lastExercise,
      showNextExercise,
      isNextButton,
      handleNextButton,
    } = this.props;
    let exerciseTitle = "";
    if (!lastExercise && nextExerciseName.toLowerCase() === "rest") {
      exerciseTitle = "";
    } else if (!lastExercise) {
      exerciseTitle = "NEXT EXERCISE:";
    } else {
      exerciseTitle = "LAST EXERCISE";
    }

    console.log(showNextExercise, "<><><>");
    return (
      <View style={styles.pauseButtonRow}>
        <View style={styles.pauseButtonContainer}>
          <TouchableOpacity onPress={handlePause} style={styles.pauseButton}>
            <Icon name="pause" size={15} color={colors.black} />
            <Text style={styles.pauseButtonText}>PAUSE</Text>
          </TouchableOpacity>
        </View>
        {showNextExercise && (
          <View
            style={[
              styles.nextExerciseContainer,
              { width: isNextButton ? width / 3 : width / 1.5 },
            ]}
          >
            {exerciseTitle !== "" && (
              <Text style={styles.nextExercise}>{exerciseTitle}</Text>
            )}

            <Text
              numberOfLines={1}
              ellipsizeMode="tail"
              style={styles.nextExerciseName}
            >
              {nextExerciseName.toLowerCase() === "rest" && "NEXT"}{" "}
              {nextExerciseName.toUpperCase()}
            </Text>
          </View>
        )}
        {isNextButton && (
          <View style={styles.pauseButtonContainer}>
            <TouchableOpacity
              onPress={handleNextButton}
              style={styles.pauseButton}
            >
              <Text style={styles.pauseButtonText}>NEXT</Text>
              <Icon
                name="chevron-right"
                size={15}
                color={colors.themeColor.color}
              />
            </TouchableOpacity>
          </View>
        )}
      </View>
    );
  }
}

PauseButtonRow.propTypes = {
  nextExerciseName: PropTypes.string.isRequired,
  handlePause: PropTypes.func.isRequired,
  lastExercise: PropTypes.bool,
  showNextExercise: PropTypes.bool.isRequired,
};

PauseButtonRow.defaultProps = {
  lastExercise: false,
  showNextExercise: false,
};

const styles = StyleSheet.create({
  pauseButtonRow: {
    width,
    flexDirection: "row",
    justifyContent: "space-between",
    backgroundColor: colors.grey.light,
    padding: 5,
    // paddingTop: 0,
  },
  pauseButtonContainer: {
    width: 117.5,
    height: 40,
  },
  pauseButton: {
    width: 117.5,
    height: 40,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: colors.white,
    paddingLeft: 15,
    paddingRight: 15,
    borderWidth: 2,
    borderColor: colors.themeColor.color,
    borderRadius: 4,
    shadowColor: colors.charcoal.light,
    shadowOpacity: 0.3,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 2,
  },
  pauseButtonText: {
    fontFamily: fonts.bold,
    fontSize: 14,
    color: colors.black,
    marginTop: 4,
    marginLeft: 5,
    marginRight: 5,
  },
  nextExerciseContainer: {
    justifyContent: "center",
    alignItems: "center",
    padding: 5,
    paddingTop: 8,
  },
  nextExercise: {
    fontFamily: fonts.standard,
    fontSize: 10,
    marginLeft: -wp("2%"),
  },
  nextExerciseName: {
    fontFamily: fonts.bold,
    fontSize: 12,
    marginRight: 5,
  },
});
