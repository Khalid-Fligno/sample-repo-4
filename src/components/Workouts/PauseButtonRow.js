import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, TouchableOpacity, StyleSheet, Dimensions } from 'react-native';
import Icon from '../../components/Shared/Icon';
import fonts from '../../styles/fonts';
import colors from '../../styles/colors';

const { width } = Dimensions.get('window');

export default class PauseButtonRow extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }
  render() {
    const { nextExerciseName, handlePause, lastExercise } = this.props;
    return (
      <View style={styles.pauseButtonRow}>
        <View style={styles.pauseButtonContainer}>
          <TouchableOpacity
            onPress={handlePause}
            style={styles.pauseButton}
          >
            <Icon
              name="pause"
              size={15}
              color={colors.coral.standard}
            />
            <Text style={styles.pauseButtonText}>
              PAUSE
            </Text>
          </TouchableOpacity>
        </View>
        <View style={styles.nextExerciseContainer}>
          <Text style={styles.nextExercise}>{!lastExercise ? 'NEXT EXERCISE:' : 'LAST EXERCISE'}</Text>
          <Text
            numberOfLines={1}
            ellipsizeMode="tail"
            style={styles.nextExerciseName}
          >
            {nextExerciseName.toUpperCase()}
          </Text>
        </View>
      </View>
    );
  }
}

PauseButtonRow.propTypes = {
  nextExerciseName: PropTypes.string.isRequired,
  handlePause: PropTypes.func.isRequired,
  lastExercise: PropTypes.bool,
};

PauseButtonRow.defaultProps = {
  lastExercise: false,
};

const styles = StyleSheet.create({
  pauseButtonRow: {
    width,
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: colors.grey.light,
    padding: 5,
    // paddingTop: 0,
  },
  pauseButtonContainer: {
    width: 117.5,
    height: 40,
  },
  pauseButton: {
    width: 117.5,
    height: 40,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.white,
    paddingLeft: 15,
    paddingRight: 15,
    borderWidth: 2,
    borderColor: colors.coral.standard,
    borderRadius: 4,
    shadowColor: colors.charcoal.light,
    shadowOpacity: 0.3,
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 2,
  },
  pauseButtonText: {
    fontFamily: fonts.bold,
    fontSize: 14,
    color: colors.coral.standard,
    marginTop: 4,
    marginLeft: 5,
    marginRight: 5,
  },
  nextExerciseContainer: {
    width: width - 117.5,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 5,
    paddingTop: 8,
  },
  nextExercise: {
    fontFamily: fonts.standard,
    fontSize: 10,
  },
  nextExerciseName: {
    fontFamily: fonts.bold,
    fontSize: 12,
    marginRight: 5,
  },
});
