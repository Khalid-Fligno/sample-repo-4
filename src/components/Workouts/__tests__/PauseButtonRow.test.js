import React from 'react';
import renderer from 'react-test-renderer';
import PauseButtonRow from '../PauseButtonRow';

describe('Pause Button Row', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PauseButtonRow
      nextExerciseName="Test"
      handlePause={jest.fn()}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
