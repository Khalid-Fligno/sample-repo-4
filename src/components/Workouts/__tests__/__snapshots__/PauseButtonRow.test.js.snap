// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pause Button Row renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(234, 234, 235)",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 5,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "height": 40,
        "width": 117.5,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(255, 255, 255)",
          "borderColor": "rgb(238, 48, 66)",
          "borderRadius": 4,
          "borderWidth": 2,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
          "shadowColor": "rgb(65, 64, 66)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
          "width": 117.5,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "rgb(238, 48, 66)",
            "fontFamily": "GothamBold",
            "fontSize": 14,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 4,
          }
        }
      >
        PAUSE
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 5,
        "paddingTop": 8,
        "width": 632.5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "GothamBook",
          "fontSize": 10,
        }
      }
    >
      NEXT EXERCISE:
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "fontFamily": "GothamBold",
          "fontSize": 12,
          "marginRight": 5,
        }
      }
    >
      TEST
    </Text>
  </View>
</View>
`;
